Replace or remove calls to `fmt.Print` (and variants)

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

<!--- Please leave a helpful description of the feature request here. --->

Random calls to `fmt.Printf` (and similar) in the code should be replaced with `log.Printf` or removed:

```console
% find aws/ -name '*.go' -print | xargs grep 'fmt\.Print'
aws//internal/vault/sdk/helper/jsonutil/json_test.go:		fmt.Printf("decoding err: %v\n", err)
aws//resource_aws_api_gateway_rest_api.go:		fmt.Printf("error normalizing policy JSON: %s\n", err)
aws//resource_aws_lakeformation_permissions_test.go:			fmt.Print(out)
aws//resource_aws_lb_ssl_negotiation_policy_test.go:			fmt.Printf("[ERROR] Problem describing load balancer policy '%s': %s", policyName, err)
aws//resource_aws_cloudwatch_event_target_test.go:		fmt.Printf("%#v", rs.Primary.Attributes)
aws//resource_aws_sns_topic_subscription_test.go:		fmt.Println(err)
```

Consider adding a linter rule to prevent these in future.

--- a/.semgrep.yml
+++ b/.semgrep.yml
@@ -415,3 +415,21 @@ rules:
           metavariable: '$Z'
           regex: '.*Pages$'
     severity: WARNING
+
+  - id: calling-fmt.Print-and-variants
+    languages: [go]
+    message: Do not call `fmt.Print` and variant
+    paths:
+      exclude:
+        - awsproviderlint/vendor/
+      include:
+        - aws/
+    patterns:
+      - pattern-either:
+        - pattern: |
+            fmt.Print(...)
+        - pattern: |
+            fmt.Printf(...)
+        - pattern: |
+            fmt.Println(...)
+    severity: WARNING

--- a/aws/internal/vault/sdk/helper/jsonutil/json_test.go
+++ b/aws/internal/vault/sdk/helper/jsonutil/json_test.go
@@ -2,7 +2,6 @@ package jsonutil
 
 import (
 	"bytes"
-	"fmt"
 	"reflect"
 	"testing"
 )
@@ -14,7 +13,7 @@ func TestJSONUtil_DecodeJSONFromReader(t *testing.T) {
 
 	err := DecodeJSONFromReader(bytes.NewReader([]byte(input)), &actual)
 	if err != nil {
-		fmt.Printf("decoding err: %v\n", err)
+		t.Errorf("decoding err: %v", err)
 	}
 
 	expected := map[string]interface{}{

--- a/aws/resource_aws_api_gateway_rest_api.go
+++ b/aws/resource_aws_api_gateway_rest_api.go
@@ -372,7 +372,7 @@ func resourceAwsApiGatewayRestApiRead(d *schema.ResourceData, meta interface{}) 
 	// I'm not sure why it needs to be wrapped with double quotes first, but it does
 	normalized_policy, err := structure.NormalizeJsonString(`"` + aws.StringValue(api.Policy) + `"`)
 	if err != nil {
-		fmt.Printf("error normalizing policy JSON: %s\n", err)
+		return fmt.Errorf("error normalizing policy JSON: %w", err)
 	}
 	policy, err := strconv.Unquote(normalized_policy)
 	if err != nil {

--- a/aws/resource_aws_cloudwatch_event_target_test.go
+++ b/aws/resource_aws_cloudwatch_event_target_test.go
@@ -674,8 +674,6 @@ func testAccAWSCloudWatchEventTargetImportStateIdFunc(resourceName string) resou
 			return "", fmt.Errorf("Not found: %s", resourceName)
 		}
 
-		fmt.Printf("%#v", rs.Primary.Attributes)
-
 		return fmt.Sprintf("%s/%s/%s", rs.Primary.Attributes["event_bus_name"], rs.Primary.Attributes["rule"], rs.Primary.Attributes["target_id"]), nil
 	}
 }

--- a/aws/resource_aws_lakeformation_permissions_test.go
+++ b/aws/resource_aws_lakeformation_permissions_test.go
@@ -256,9 +256,8 @@ func testAccCheckAWSLakeFormationPermissionsDestroy(s *terraform.State) error {
 			},
 		}
 
-		out, err := conn.ListPermissions(input)
+		_, err := conn.ListPermissions(input)
 		if err == nil {
-			fmt.Print(out)
 			return fmt.Errorf("Resource still registered: %s %s", catalogId, principal)
 		}
 	}

--- a/aws/resource_aws_lb_ssl_negotiation_policy_test.go
+++ b/aws/resource_aws_lb_ssl_negotiation_policy_test.go
@@ -2,6 +2,7 @@ package aws
 
 import (
 	"fmt"
+	"log"
 	"testing"
 
 	"github.com/aws/aws-sdk-go/aws"
@@ -145,7 +146,7 @@ func testAccCheckLBSSLNegotiationPolicy(elbResource string, policyResource strin
 		})
 
 		if err != nil {
-			fmt.Printf("[ERROR] Problem describing load balancer policy '%s': %s", policyName, err)
+			log.Printf("[ERROR] Problem describing load balancer policy '%s': %s", policyName, err)
 			return err
 		}
 

--- a/aws/resource_aws_sns_topic_subscription_test.go
+++ b/aws/resource_aws_sns_topic_subscription_test.go
@@ -582,10 +582,10 @@ func testAccCheckAWSSNSTopicSubscriptionRedrivePolicyAttribute(attributes map[st
 const awsSNSPasswordObfuscationPattern = "****"
 
 // returns the endpoint with obfuscated password, if any
-func obfuscateEndpoint(endpoint string) string {
+func obfuscateEndpoint(t *testing.T, endpoint string) string {
 	res, err := url.Parse(endpoint)
 	if err != nil {
-		fmt.Println(err)
+		t.Errorf("error parsing URL: %s", err)
 	}
 
 	var obfuscatedEndpoint = res.String()
@@ -607,7 +607,7 @@ func TestObfuscateEndpointPassword(t *testing.T) {
 		"https://username:password@example.com/myroute": "https://username:****@example.com/myroute",
 	}
 	for endpoint, expected := range checks {
-		out := obfuscateEndpoint(endpoint)
+		out := obfuscateEndpoint(t, endpoint)
 
 		if expected != out {
 			t.Fatalf("Expected %v, got %v", expected, out)

